### PR TITLE
Improve email templates

### PR DESCRIPTION
**Pull request by kingdish**
### Proposed Changes:
Various fix and improvements per comments in #7 
1. Divide template into partials
2. Create external css for template
3. Automatically determine conference / organization / default logo. Add corresponding tests
4. Add alt text for logo
5. Display conference color for header and footer. Add corresponding tests.
6. Refactor `mailbot.rb`
7. Style fix

### Contributors:
@kingdish Jimmy
@rajavi-mishra Rajavi

### CODE REVIEW

**Review by CactusPuppy — APPROVED** [review]
LGTM

**Review by zfred219 — COMMENTED** [review]
Overall looks pretty good, especially in structure improvement (DRYness, like interaction between different controller method and same partial view) and I think it should be highly-readable and scalable code with decent ABC score.

Maybe consider add cucumber/any behavior-driven tests to mock user's behavior as well as proceed more integration-level tests.

File: app/assets/stylesheets/mailbot.css [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  h1,
+  h3,
+  h4,
+  h5,
+  h6 {
+      font-weight: 400;
+      line-height: 1.3;
+  }
+  p {
+    color: #0B3559;
```
Comment (zfred219) [inline app/assets/stylesheets/mailbot.css:24]: CSS is pretty neat.  (Maybe consider use variable for color)

File: app/assets/stylesheets/mailbot.css [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    font-weight: 400;
+    line-height: 2;
+  }
+
+  #border {
+    background-color: #0B3559;
```
Comment (zfred219) [inline app/assets/stylesheets/mailbot.css:30]: CSS is pretty neat.  (Maybe consider use variable for color)

**Comment by EnjiaXu** [comment]
**What is the goal of this PR?**
Improving the design/code structure of an existing mailing feature.

**Were you able to view screenshots, story cards, and or test the app?**
We were able to go to the [heroku deployment](https://cs169l-sp2021-snapcon.herokuapp.com) of the forked repo and look around, but we were not able to interact/play with the mailing feature itself. 

**Is the PR in a shape that makes it easy to review?**
Yes. The code changes are summarized clearly in the list. It also provides context by giving a link to the previous PR regarding the same feature.

**What are the status of the tests?**
The tests included are all passing. For this PR specifically, it was mostly modifying/refactoring existing code (including some testing code), so there are not many new tests being added (which makes sense). After scanning through the spec folder, it seems like the existing tests are well-covered. However, it does not seem to use the cucumber gem, so we are unsure how the behavior/integration testings are done (maybe just capybara with rspec?).

**What is the style of the code like?**
They have linters (rubocop) that catch their code style problems, they are all passing.

**Is there anything missing from this PR that should be included?**
For this PR, since it's mostly code structure changes, the files changed are quite self-explanatory.

**Feature Design: Are there any parts of this feature that are cause for concern?**
There does not seem to have such parts that can cause abusable problems. The overall restructuring makes the code look cleaner and less redundant.

---

### Overall Assessment

**What does this PR do well?**
Provides clear objectives and contexts. The problems highlighted in the previous PR are managed properly.

**What is something that could be improved with this PR?**
Maybe adding one or two screenshots of the features being added/modified can make it even easier to understand to an outsider, but they can be hard to include for a mailing feature.

File: app/assets/stylesheets/mailbot.css [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    padding: 25px;
+    color:#fff;
+  }
+
+  #content {
+    background-color: #fff;
+    padding: 100px;
+    color: #0B3559;
```
Comment (zfred219) [inline app/assets/stylesheets/mailbot.css:38]: CSS is pretty neat overall. 
 (Maybe consider use variable for color)

File: app/helpers/conference_helper.rb [hunk 4]
```diff
@@ -1,5 +1,8 @@
 # frozen_string_literal: true
```
Comment (zfred219) [inline app/helpers/conference_helper.rb:1]: Maybe consider remove this code comment for readability.

File: app/mailers/mailbot.rb [hunk 5]
```diff
@@ -5,21 +5,25 @@
 YTLF_TICKET_ID = 50
 
 class Mailbot < ActionMailer::Base
+  helper ConferenceHelper
+
+  default bcc:           -> { SNAPCON_BCC_ADDRESS },
+          template_name: -> { EMAIL_TEMPLATE },
+          to:            -> { @user.email },
+          from:          -> { @conference.contact.email }
+
```
Comment (EnjiaXu) [inline app/mailers/mailbot.rb:14]: The default usage here makes the code a lot cleaner than before!

**Pull request merged**